sklearn/utils/estimator_checks.py

```diff
@@ -112,7 +112,11 @@ def _yield_checks(estimator):
             # We skip pairwise because the data is not pairwise
             yield check_sample_weights_shape
             yield check_sample_weights_not_overwritten
-            yield check_sample_weight_equivalence
+            for sparse_container in [None] + CSR_CONTAINERS:
+                yield partial(
+                    check_sample_weight_equivalence,
+                    sparse_container=sparse_container,
+                )
```

Comment:
We should yield the `csr_array` call under an `if SPARSE_ARRAY_PRESENT and tags.input_tags.sparse` block.